server/repository/src/database_settings.rs

```diff
@@ -111,8 +111,52 @@ impl diesel::r2d2::CustomizeConnection<diesel::SqliteConnection, diesel::r2d2::E
 pub fn get_storage_connection_manager(settings: &DatabaseSettings) -> StorageConnectionManager {
     let connection_manager =
         ConnectionManager::<DBBackendConnection>::new(&settings.connection_string());
-    let pool = Pool::new(connection_manager).expect("Failed to connect to database");
-    StorageConnectionManager::new(pool)
+
+    // Note: the build() call isn't failing when you have an incorrect server or database name
+    // which is why the connect() call is made
+    match connection_manager.connect() {
+        Ok(_conn) => {
```

Comment:
concerned that I couldn't find a 'disconnect' method for the connection 🤷

Reply:
The connection is managed using Rust's `Drop` trait, e.g when the connection goes out of scope, it will be disconnected automatically.
https://doc.rust-lang.org/rust-by-example/trait/drop.html

TBH, I'm kind of surprised there's no explicit disconnect method that could be optionally run. I guess if they were to allow that, it would be hard for the rust compiler to recognise if the connection should and/or could be freed.

For my own interest, I think this is the actual code in postgres land...
https://github.com/diesel-rs/diesel/blob/50c2c85032857a65e4a8d68dc851fa7ab91d65b2/diesel/src/pg/connection/raw.rs#L141